### PR TITLE
Controller Runtime test suite improvements (on top of #3773)

### PR DESCRIPTION
**Pull request by sunjayBhatia**

### CODE REVIEW

**Comment by skriss** [comment]
ready for a rebase

File: internal/controller/gatewayclass_test.go [hunk 1]
```diff
@@ -65,15 +57,15 @@ var _ = Describe("GatewayClass Controller", func() {
 			}, timeout, interval).Should(Succeed())
 
 			By("Expecting to delete finish")
```
Comment (skriss) [inline internal/controller/gatewayclass_test.go:59]:
```suggestion
			By("Expecting delete to finish")
```

File: internal/controller/gatewayclass_test.go [hunk 2]
```diff
@@ -103,15 +95,15 @@ var _ = Describe("GatewayClass Controller", func() {
...
```
Comment (skriss) [inline internal/controller/gatewayclass_test.go:97]:
```suggestion
			By("Expecting delete to finish")
```

File: internal/controller/gatewayclass_test.go [hunk 3]
```diff
@@ -161,10 +153,10 @@ var _ = Describe("GatewayClass Controller", func() {
...
```
Comment (skriss) [inline internal/controller/gatewayclass_test.go:155]:
```suggestion
			By("Expecting delete to finish")
```

File: internal/controller/gatewayclass_test.go [hunk 4]
```diff
@@ -175,15 +167,15 @@ var _ = Describe("GatewayClass Controller", func() {
...
```
Comment (skriss) [inline internal/controller/gatewayclass_test.go:169]:
```suggestion
			By("Expecting delete to finish")
```

File: internal/controller/suite_test.go [hunk 5]
```diff
@@ -44,13 +43,10 @@ var (
 
 func TestAPIs(t *testing.T) {
 	RegisterFailHandler(Fail)
-
-	RunSpecsWithDefaultAndCustomReporters(t,
-		"Runtime Controller Suite",
-		[]Reporter{printer.NewlineReporter{}})
+	RunSpecs(t, "Runtime Controller Suite")
```
Comment (skriss) [inline internal/controller/suite_test.go:46]:
```suggestion
	RunSpecs(t, "Controller Runtime Suite")
```

**Review by skriss — CHANGES_REQUESTED** [review]
noticed a couple other pre-existing nits to fix, otherwise LGTM

**Review by skriss — APPROVED** [review]

**Review by stevesloka — APPROVED** [review]

**Pull request merged**